src/gui_common/SegmentedBar.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   A ProgressBar that is split up into IconProgressBars, data is stored in a dictionary
+/// </summary>
+public class SegmentedBar : ProgressBar
+{
+    public PackedScene IconProgressBarScene = GD.Load<PackedScene>("res://src/gui_common/IconProgressBar.tscn");
+    public string Type;
+
+    public void BarToggled(InputEvent @event, IconProgressBar bar)
+    {
+        if (@event is InputEventMouseButton eventMouse && @event.IsPressed())
+        {
+            IconBarConfig config = new IconBarConfig(bar);
+            config.Disabled = !config.Disabled;
+            new SegmentedBarConfig(this).handleBarDisabling(bar);
```

Comment:
My initial reaction to this is "what?".
I think this is an artifact of the fact that the configs should really be stored in this class and not somewhere else. Though the editor GUI should also probably hang onto them so that it can then update stuff, but that should also go through a method, so the config class is only like an identifier which bar to update.